MinMaxOfArray.java

```diff
@@ -0,0 +1,127 @@
+// Time Complexity : O(log n), for diving the problem in 2 equal halves
+// Space Complexity : O(log n), for the stack space
+// I'm not sure about how to calculate total comparisions in this method
```

Comment:
In the recursive case, you are doing 3 comparisons in each call, so it approximates to 3(n / 2) - number of elements is halved at every call.